src/egon/data/datasets/gas_grid.py

```diff
@@ -406,10 +406,16 @@ def insert_gas_pipeline_list(gas_nodes_list):
         ]
     )
 
+    buses = tuple(db.select_dataframe(
+        f"""SELECT bus_id FROM grid.egon_etrago_bus
+            WHERE scn_name = '{scn_name}' AND country = 'DE';
+        """
+    )['bus_id'])
+
     # Insert data to db
     db.execute_sql(
         f"""DELETE FROM grid.egon_etrago_link WHERE "carrier" = 'CH4' AND
-           scn_name = '{scn_name}';
+           scn_name = '{scn_name}' AND bus0 IN {buses} AND bus1 IN {buses};

```

Comment:
In general you shouldn't exchange data from SQL to python variables just to use them in another SQL query. That results in very long SQL statements which could exceed the limit of allowed characters and needs more computational resources.

Reply:
Thank you for the hint, changed all queries.